Automattic/newspack-newsletters
# chore: update dependencies

### All Submissions:






### Changes proposed in this Pull Request:



Updates NPM packages and Node support to v16. Also implements Dependabot.

### How to test the changes in this Pull Request:

1. Using NVM, set your current Node version to v16.11.1 (matching Calypso's supported version)
2. Ensure `npm ci` runs
3. Ensure all NPM build, lint and test scripts still run (`build`, `dev`, `lint`, `test`)
4. Smoke test plugin functionality (there should be no functional changes compared to `master`)

### Other information:
